packages/plugin-ext/src/hosted/browser/plugin-worker.ts

```diff
@@ -16,26 +16,41 @@
 import { injectable } from '@theia/core/shared/inversify';
 import { Emitter } from '@theia/core/lib/common/event';
 import { RPCProtocol, RPCProtocolImpl } from '../../common/rpc-protocol';
+import { environment } from '@theia/core';
 
 @injectable()
 export class PluginWorker {
 
     private worker: Worker;
+
     public readonly rpc: RPCProtocol;
+
     constructor() {
         const emitter = new Emitter<string>();
-        this.worker = new (require('../../hosted/browser/worker/worker-main'));
-        this.worker.onmessage = message => {
-            emitter.fire(message.data);
-        };
-        this.worker.onerror = e => console.error(e);
+
+        if (this.isElectron()) {
+            console.log('PluginWorker : Frontend plugins are temporarily not supported for Electron');
+        } else {
+            require('./worker/worker-main');
```

Comment:
Why are we requiring this without using the result as before? Could you explain why you needed to change loading the worker?

Reply:
If the file is required somewhere, the webpack generates `examples/browser/lib/worker-ext.js` and `examples/electron/lib/worker-ext.js` files.
If we don't require './worker/worker-main', the 'worker-ext.js' is missing.

If somebody know how to force the webpack to generate .js file without adding that 'require', it would be helpful.

Reply:
It was reworked. After adding a webpack configuration, the javascript file is taken from the @threia/plugin-ext directly.